MapleServer2/Types/Inventory.cs

```diff
@@ -45,6 +45,11 @@ public class Inventory {
             // Item has a slot set, try to use that slot
             if (item.Slot >= 0) {
                 if (!SlotTaken(item, item.Slot)) {
+                    //Performs a item exist check before adding item
+                    if (Check(item))
```

Comment:
Does it make sense to add this `Check` method directly inside AddInternal since it seems like you'd always want to do this check before calling it?

Reply:
```suggestion
                   if (!session.Inventory.Add(item))
                {
                    session.Inventory.Update(item, (item.Amount + item.Amount));
                    session.Send(ItemInventoryPacket.Update(item.Uid, (item.Amount + item.Amount)));
                }
                else
                {
                    session.Send(ItemInventoryPacket.Add(item));
                }
```
Issue is that if I bring the `Check` method directly inside AddInternal, There is no way for the above code in `FieldEnterHandler` to know whether to run `ItemInventoryPacket.Add` or `ItemInventoryPacket.Update`

Feel free to experiment with that. I'm gonna mess with it again later tonight.